sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystQl.scala

```diff
@@ -41,16 +47,9 @@ private[sql] class CatalystQl(val conf: ParserConf = SimpleParserConf()) {
     }
   }
 
-
-  /**
-   * Returns the AST for the given SQL string.
-   */
-  protected def getAst(sql: String): ASTNode = ParseDriver.parse(sql, conf)
-
-  /** Creates LogicalPlan for a given HiveQL string. */
-  def createPlan(sql: String): LogicalPlan = {
+  protected  def safeParse[T](sql: String, ast: ASTNode)(toResult: ASTNode => T): T = {
```

Comment:
there is a extra space before the def.

can you also add some doc explaining what this function does (e.g. what "safe" means)?